tests/flows/native_token_bridge.go

```diff
@@ -0,0 +1,387 @@
+package flows
+
+import (
+	"context"
+	"crypto/ecdsa"
+	"math/big"
+
+	"github.com/ava-labs/subnet-evm/accounts/abi/bind"
+	"github.com/ava-labs/subnet-evm/core/types"
+	nativetokendestination "github.com/ava-labs/teleporter/abi-bindings/go/CrossChainApplications/NativeTokenBridge/NativeTokenDestination"
+	nativetokensource "github.com/ava-labs/teleporter/abi-bindings/go/CrossChainApplications/NativeTokenBridge/NativeTokenSource"
+	"github.com/ava-labs/teleporter/tests/interfaces"
+	"github.com/ava-labs/teleporter/tests/utils"
+	deploymentUtils "github.com/ava-labs/teleporter/utils/deployment-utils"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"github.com/ethereum/go-ethereum/log"
+	. "github.com/onsi/gomega"
+)
+
+func NativeTokenBridge(network interfaces.LocalNetwork) {
```

Comment:
same, can we use `interfaces.Network` here instead?

Reply:
It was `interfaces.Network`, but Cam suggested `interfaces.LocalNetwork`. I think that's the right decision because these tests won't be able to be run on a testnet, since they require their own subnet that has admin address set for the Native Minter precompile. So while we could spin up a subnet on Fuji to test these contracts, the flow would have to be different from what is here.

Reply:
sounds good no strong pref here. After this gets deployed and a fuji subnet has it set up, would we be able to use that subnet as a testnet if wanted?

Reply:
We could test the functionality, yeah, but we would need to not use the same setup code. So we could split them out at that point of `setup` and then `test`.